pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -405,24 +435,26 @@ def format(self) -> DocumentFormat:
             raise ValueError(f'Unknown document media type: {media_type}') from e
 
 
-@dataclass(repr=False)
+@dataclass(init=False, repr=False)
 class BinaryContent:
     """Binary content, e.g. an audio or image file."""
 
     data: bytes
     """The binary data."""
 
-    media_type: AudioMediaType | ImageMediaType | DocumentMediaType | str
-    """The media type of the binary data."""
-
     _: KW_ONLY
 
-    identifier: str | None = None
-    """Identifier for the binary content, such as a URL or unique ID.
+    media_type: AudioMediaType | ImageMediaType | DocumentMediaType | str
+    """The media type of the binary data."""
 
-    This identifier can be provided to the model in a message to allow it to refer to this file in a tool call argument, and the tool can look up the file in question by iterating over the message history and finding the matching `BinaryContent`.
+    identifier: str
```

Comment:
@DouweM @kyuam32  -- was this intended to go to `str` vs keeping `str | None`? we store old json to rehydrate later into chags and now `ModelMessageAdapter.validate_json` throws because it expects `identifier` to have a value for those old dicts

Reply:
@kousun12 Ay that was unintentional, I was relying on the fact that we always set an `identifier` in `__init__`, but that wouldn't work for validation. Can you create a new issue please?